src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs

```diff
@@ -286,7 +286,15 @@ private SslStreamCertificateContext(X509Certificate2 target, ReadOnlyCollection<
                     // Note that if server does not send OCSP staples, clients may still
                     // contact OCSP responders directly.
                     _nextDownload = DateTimeOffset.UtcNow.AddSeconds(5);
-                    _ocspExpiration = _nextDownload;
+
+                    if (_ocspExpiration < _nextDownload)
+                    {
+                        // this prevent forced download on next call if called before _nextDownload.
+                        _ocspExpiration = _nextDownload;
+
+                        // Drop the cached OCSP response, it expires in next few seconds anyway.
+                        _ocspResponse = null;
```

Comment:
prevent the forced download by expiry... but remove the response, which forces the download.

I'm not sure what the intent of this block is.

Reply:
maybe it's getting a bit tangled....

In case OCSP fetch fails, we don't want to try again immediatelly (we postpone `_nextDownload`), so `GetOcspResponseAsync` falls to the end and returns the cached response. But in degenerate case where fetches keep failing, we can't keep postponing indefinitely as once we get past `_ocspExpiration`, the first branch of `GetOcspResponseAsync` would force the new fetch immediately.

So if that happens (i.e. expiration would kick in before the next fetch), we drop the response as it would expire anyway and shift "expiration" so that the forced fetch still happens after the backoff.

Now that I am thinking, maybe keeping `_ocspExpiration` as is and changing `GetOcspResponseAsync` to do 

```cs
            if (now > _ocspExpiration)
            {
                if (now < _nextDownload) return null;
                return DownloadOcspAsync();
            }
```

would be clearer?

Reply:
I understand removing line 289; if the expiration still has several days to go, don't flush it away just because a soft refresh failed.

For soft refresh, backing off a few seconds makes sense.  I'm not sure I see the value in backing off a hard refresh (assuming it fails here, no OCSP is stapled, and now potentially many clients turn around and try to talk to that same server, so backing off might actually _increase_ the load on the server).

If you think backing off in hard refresh is important, then I'd probably go with changing GetOcspResponseAsync, and on the new if add a comment like `// If we're still backing off from a failure, just return null` to explain what that state is.

Reply:
I see, I will just remove the `_ocspExpiration` assignment then.